src/In.ProjectEKA.HipService/Discovery/PatientDiscovery.cs

```diff
@@ -56,9 +59,22 @@ public class PatientDiscovery
                     .Map(async patient =>
                     {
                         await discoveryRequestRepository.Add(new Model.DiscoveryRequest(request.TransactionId,
-                            request.Patient.Id, patient.Identifier));
+                            request.Patient.Id, patient.Identifier,
+                            patient.MaskedIdentifier));
+                        var unlinkedAccounts = GetUnlinkedCareContexts(linkedCareContexts, patient);
+                        var careContextRepresentations = unlinkedAccounts.ToList();
+                        var addToUnlinkedAccounts = AddToUnlinkedAccounts(careContextRepresentations, patient.Identifier);
+                        if (!careContextRepresentations.Any() &&
+                            !addToUnlinkedAccounts)
+                        {
+                            return (null,
+                                new ErrorRepresentation(new Error(ErrorCode.FailedToGetLinkedCareContexts,
+                                    "Unable to update unlinked care contexts")));
+                        }
+
+                        patient.Identifier = patient.MaskedIdentifier;

```

Comment:
at the time of adding it into a DiscoveryRequest table these are two different fields, but at the time of returning output to CM as response we are only passing reference number and display, so we need to replace line 75 to show that on output
<img width="442" alt="Screenshot 2020-07-09 at 10 53 22 AM" src="https://user-images.githubusercontent.com/63891910/87000326-721cbb80-c1d2-11ea-9e76-73d718afbb48.png">

Reply:
I understand the what you are doing, my question is why mutation?

Reply:
yeah we supposed to keep CM untouched, since they are reading and returning final response from CM, and they are returning reference field not a masked even, they don't even bother about it.